### PR TITLE
feat(security): Add renewal mechanism for Vault management token in security Consul bootstrapper

### PR DESCRIPTION
**Pull request by jim-wang-intel**
Changes made:
 - add renewal process in security-bootstrapper's consul entrypoint
 - add renewal process in snap package as a new simple service using script

The renewal process will be kicked off once the Consul ACLs setup is successfully done and is running as long-running process.
The Vault token renewal happens when its TTL is less than 10% of its original period if it is not expired yet.

Closes: #3335

Signed-off-by: Jim Wang <yutsung.jim.wang@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the current behavior?

No renewal of token

## What is the new behavior?
Renewal for token process added

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

## New Imports

- [ ] Yes
- [x ] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information
Locally tested and observed logging message something like `"successfully renewed"`:
```
......
Mon Apr  5 16:16:22 UTC 2021 try to renew management token with http://edgex-vault:8200/v1/auth/token/renew-self:
Mon Apr  5 16:16:22 UTC 2021   successfully renewed management token
Mon Apr  5 16:16:22 UTC 2021 checking Vault management token's TTL with http://edgex-vault:8200/v1/auth/token/lookup-self ...
Mon Apr  5 16:16:23 UTC 2021 management token will be renewed after 161 seconds
......
```

### CODE REVIEW

File: cmd/security-bootstrapper/entrypoint-scripts/consul_wait_install.sh [hunk 1]
```diff
@@ -91,7 +135,22 @@ if [ "${ENABLE_REGISTRY_ACL}" == "true" ]; then
...
   if [ "${setupACL_code}" -ne 0 ]; then
     echo "$(date) failed to set up Consul ACL"
+  else
+    # kick off the token renewal process
+    echo "$(date) launching vault management token renewal process:"
+    if [ -f "${STAGEGATE_REGISTRY_ACL_SECRETSADMINTOKENPATH}" ]; then
+      mgmt_token=$(jq -r '.auth.client_token' "${STAGEGATE_REGISTRY_ACL_SECRETSADMINTOKENPATH}")
+      parsing_token_ret_code=$?
+      if [ "${parsing_token_ret_code}" -eq 0 ]; then
+        renewToken "${mgmt_token}" &
```
Comment (lenny-intel) [inline cmd/security-bootstrapper/entrypoint-scripts/consul_wait_install.sh:145]: How is this running it as a different process?

File: snap/local/runtime-helpers/bin/renew_mgmt_token.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
+#!/bin/bash -e
```
Comment (lenny-intel) [inline snap/local/runtime-helpers/bin/renew_mgmt_token.sh:1]: Couldn't we reuse this script and call it from the consul wait script. Obviously the Docker file would have to copy it to the folder that is volume mounted.

File: cmd/security-bootstrapper/entrypoint-scripts/consul_wait_install.sh (hunk 1 above)
Comment (jim-wang-intel) [reply cmd/security-bootstrapper/entrypoint-scripts/consul_wait_install.sh:145]: the "&" symbol make it run and fork as a background process and thus it is different from the main process we are running in this entry point.

File: snap/local/runtime-helpers/bin/renew_mgmt_token.sh (hunk 2 above)
Comment (jim-wang-intel) [reply snap/local/runtime-helpers/bin/renew_mgmt_token.sh:1]: never seen done this way before; i thought snap is a separate world by itself. we haven't done sharing a file between a snap and a docker

Comment (lenny-intel) [reply snap/local/runtime-helpers/bin/renew_mgmt_token.sh:1]: This script is duplicate code for what you have in the consul wait script so the Docker file just needs to copy this file to the shared location next to the consul wait script.

File: cmd/security-bootstrapper/entrypoint-scripts/consul_wait_install.sh (hunk 1 above)
Comment (lenny-intel) [reply cmd/security-bootstrapper/entrypoint-scripts/consul_wait_install.sh:145]: Ahh,  ok. did see that....

File: snap/local/runtime-helpers/bin/renew_mgmt_token.sh (hunk 2 above)
Comment (jim-wang-intel) [reply snap/local/runtime-helpers/bin/renew_mgmt_token.sh:1]: they are not exactly the same: 
- the snap is running as systemd service and thus I remove all the date-time logging information from the script eg. $(date)
- the renewToken in snap is not running as background process as it is running as a separate service here.

so we cannot just copy the exact file.

Comment (lenny-intel) [reply snap/local/runtime-helpers/bin/renew_mgmt_token.sh:1]: But, will little refactoring it could be. It is 99% duplicate code.

Comment (jim-wang-intel) [reply snap/local/runtime-helpers/bin/renew_mgmt_token.sh:1]: not sure how you can do refactoring?  remember every line of `echo $(date) .....` is different from the two.

Comment (lenny-intel) [reply snap/local/runtime-helpers/bin/renew_mgmt_token.sh:1]: I am sure you can figure out a way ;-)

Comment (jim-wang-intel) [reply snap/local/runtime-helpers/bin/renew_mgmt_token.sh:1]: it is kind of kludge solution refactoring either way IMO.  I think the better solution is to code this function in a security-bootstrapper subcommand and both snap and docker world can share. I'll rework on that to reduce the duplication concern you have.

**Comment by jim-wang-intel** [comment]
This PR is no longer needed as the latest approach will be using dynamic generation of consul-token on-the-fly through each service's token.

**Pull request closed without merging**